packages/traffic_light/src/traffic_light_node.py

```diff
@@ -1,64 +1,66 @@
+        ~activation_order (:obj:`list` of :obj:`int`): The order of activation of the
+            LEDs, default is [0,1,2,3]
+        ~green_time (:obj:`float`): Duration of the green signal in seconds, default is 5
+        ~all_red_time (:obj:`float`): Duration of a signal when all direction are red (waiting
+            for the Duckiebots to clear the intersection) in seconds, default is 4
+        ~frequency (:obj:`float`): The blinking frequency of the green signal, default is 7.8
+
     """
 
-    def __init__(self):
-        self.node_name = rospy.get_name()
-        rospy.loginfo("[%s] Initializing..." % (self.node_name))
+    def __init__(self, node_name):
+        # Initialize the DTROS parent class
+        super(TrafficLightNode, self).__init__(node_name=node_name)
 
         # Import protocols
-        self.LED_protocol = rospy.get_param("~LED_protocol")
-        self.TL_protocol = rospy.get_param("~TL_protocol")
-        self.scale = rospy.get_param("~LED_scale")
+        self.parameters["~number_leds"] = None
+        self.parameters["~activation_order"] = None
+        self.parameters["~green_time"] = None
+        self.parameters["~all_red_time"] = None
+        self.parameters["~frequency"] = None
+        self.updateParameters()
+
+        self.green_idx = 0
 
-        # Setup parameters
-        self.green_frequency = self.LED_protocol['signals']['TL_GO']['frequency']
-        self.green_light_duration = self.TL_protocol['timings']['green_time']
-        self.all_red_duration = self.TL_protocol['timings']['all_red_time']
-        self.number_leds = self.TL_protocol['leds']['number_leds']
-        self.activation_order = self.TL_protocol['leds']['activation_order']
+        cycle_duration = self.parameters["~green_time"] + self.parameters["~all_red_time"]
 
+        # Create the color mask
         self.color_mask = [0]*5
-        self.color_mask[0:self.number_leds-1] = [1]*self.number_leds
-        self.green_idx = 0
+        self.color_mask[0:self.parameters["~number_leds"]-1] = [1]*self.parameters["~number_leds"]
 
-        cycle_duration = self.green_light_duration + self.all_red_duration
+        # Function mapping to LEDEmitterNode's `set_custom_pattern` service
+        self.changePattern = rospy.ServiceProxy(rospy.get_namespace()+'led_emitter_node/set_custom_pattern',
+                                                SetCustomLED)
 
-        # Publishers
-        self.pub_pattern = rospy.Publisher("~custom_pattern",
-                                           LEDPattern,
-                                           queue_size=1)
-        # Initialize timer for direction change
+        # Start a timer that will regularly call a method that changes
+        # the direction that get green light
         self.traffic_cycle = rospy.Timer(rospy.Duration(cycle_duration),
                                          self.change_direction)
-        # Scale intensity of the LEDs & convert to right order
```

Comment:
Are you dimming the lights somewhere else?

Reply:
Yes, it uses the LEDEmitterNode and it handles the scaling. I have forgotten to remove this comment. Will handle it now.